### DISCUSSION TITLE
'Failed to parse ' error from FlightControl/MSSP API

### DISCUSSION THREAD

**Discussion by cozadk** [CrowdStrike/falconpy#143] [Q&A] [answered]
I've started exploring the Flight Control Service Class and am running into parsing errors with the endpoints that take list inputs.  I've double checked and don't believe that there is a problem with the formatting/encoding of the parameter values but the URLs as returned in the error message include no-width spaces.

`https://api.crowdstrike.com\u200b/mssp\u200b/entities\u200b/children\u200b/v1?ids=<xxxxxxxxx>`

This happens with `falcon.getChildren` and `falcon.getCIDGroupById` for example and is the case whether there are multiple values in the list or only a single CID or groupID.  My calls using `falcon.queryChildren` are working as expected.

Any help is much appreciated.

Thanks,
Kyle

**Comment by jshcodes** [comment]
Hi Kyle!

I'm working on recreating this now.

**Reply by jshcodes** [reply]
I've successfully recreated this, and am leaning towards confirming it as a bug. I will create an issue for it once I have a little more detail regarding the root cause. (I'm working on this now.)  We will target to have the fix as part of our next merge. (So it will release in __version 0.4.8__.)

In the interim, I believe this workaround will get you the detail you need as the Uber class does not appear to have the issue.  (Please note: For this example I'm just pulling all of the children and then feeding the list to the getChildren function.)

```python
from falconpy.api_complete import APIHarness as Uber
falcon = Uber(creds={"client_id": client_id, "client_secret": client_secret})
kids = falcon.command("queryChildren")["body"]["resources"]
result = falcon.command("getChildren", ids=kids)
```

Let me know if you see any more issues with the MSSP service class.  It's recently released, the feedback is truly helpful! :smile:

**Reply by cozadk** [reply]
Thanks for the quick response!  Appreciate the workaround and will definitely let you know if I run into any other issues

**Comment by jshcodes** [comment]
This is a confirmed bug and is now being tracked here: Issue #144 

We have a fix that we are unit testing now and will include in the __0.4.8__ release.

**Comment by jshcodes** [comment, marked as answer]
Hello!

This should be resolved by our most recent update. If you are still experiencing this issue, please upgrade to the latest version (__0.4.8__) to address.  
```shell
python3 -m pip install crowdstrike-falconpy --upgrade
```

Let us know if you encounter anything else.  :smile:
